GiniVision/Classes/Core/GiniScreenAPICoordinator.swift

```diff
@@ -32,18 +32,22 @@ internal final class GiniScreenAPICoordinator: NSObject, Coordinator {
     var cameraViewController: CameraViewController?
     var imageAnalysisNoResultsViewController: ImageAnalysisNoResultsViewController?
     var reviewViewController: ReviewViewController?
-    var multiPageReviewController: MultipageReviewViewController?
+    lazy var multiPageReviewViewController: MultipageReviewViewController = {
```

Comment:
Yes. It also concerned me but I think it makes totally sense since it is probably the screen with more interaction. Also I think that in terms of performance is much worse to create the view controller every time.

Reply:
Then you have to be really careful to clear the state after you're done with the controller. Because it would be really bad if you use the same one afterward but it still has some data from the previous screen.

And again, it seems like premature optimisation.

Reply:
You will never have documents from previous session. When using this (screen API), you can only go back to multipage screen if you had a problem in the analysis. Which makes sense, since you want to have the same documents. Once the analysis has been completed, the Screen API should be deallocated.

Reply:
Well, if you like it this way keep it.
From my experience it brings more problems than solutions, but it's not a big deal.